src/services/CertificateGenerationService.ts

```diff
@@ -290,17 +291,24 @@ class CertificateGenerationService {
    * @returns the signature as a base64 encoded string
    */
   public async getSignature(staffId: string): Promise<string | null> {
-    return this.s3Client
-      .download(`cvs-signature-${process.env.BUCKET}`, `${staffId}.base64`)
-      .then((result: GetObjectOutput) => {
-        return result.Body!.toString();
-      })
-      .catch((error: ServiceException) => {
-        console.error(
-          `Unable to fetch signature for staff id ${staffId}. ${error.message}`
-        );
-        return null;
-      });
+    try {
+      const result: GetObjectOutput = await this.s3Client
+          .download(`cvs-signature-${process.env.BUCKET}`, `${staffId}.base64`);
+
+      if (result.Body instanceof Readable) {
+        const chunks: Uint8Array[] = [];
+        for await (const chunk of result.Body) {
+          chunks.push(chunk);
+        }
+        const buffer = Buffer.concat(chunks);
+        return buffer.toString("utf-8");
+      } else {
+        console.log(`Unexpected body type: ${typeof result.Body}`);
```

Comment:
This might be better to throw and raise the error in the catch block below - although we have technically fetched the signature.

If not in the block below, I think the staff id would definitely useful information to have in this log.